agent/ec2/ec2_metadata_client.go

```diff
@@ -15,25 +15,30 @@ package ec2
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io/ioutil"
 	"net"
-	"net/http"
-	"strings"
+	net_http "net/http"
 	"time"
 
+	"github.com/aws/amazon-ecs-agent/agent/ec2/http"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
 )
 
 const (
-	EC2_METADATA_SERVICE_URL                      = "http://169.254.169.254"
-	SECURITY_CREDENTIALS_RESOURCE                 = "/2014-02-25/meta-data/iam/security-credentials/"
-	INSTANCE_IDENTITY_DOCUMENT_RESOURCE           = "/2014-02-25/dynamic/instance-identity/document"
-	INSTANCE_IDENTITY_DOCUMENT_SIGNATURE_RESOURCE = "/2014-02-25/dynamic/instance-identity/signature"
-	SIGNED_INSTANCE_IDENTITY_DOCUMENT_RESOURCE    = "/2014-02-25/dynamic/instance-identity/pkcs7"
-	EC2_METADATA_REQUEST_TIMEOUT                  = time.Duration(1 * time.Second)
+	EC2MetadataServiceURL                     = "http://169.254.169.254"
+	SecurityCrednetialsResource               = "/2014-02-25/meta-data/iam/security-credentials/"
+	InstanceIdentityDocumentResource          = "/2014-02-25/dynamic/instance-identity/document"
+	InstanceIdentityDocumentSignatureResource = "/2014-02-25/dynamic/instance-identity/signature"
+	SignedInstanceIdentityDocumentResource    = "/2014-02-25/dynamic/instance-identity/pkcs7"
+
+	macResource            = "/2014-02-25/meta-data/mac"
+	vpcIDResourceFormat    = "/2014-02-25/meta-data/network/interfaces/macs/%s/vpc-id"
+	subnetIDResourceFormat = "/2014-02-25/meta-data/network/interfaces/macs/%s/subnet-id"
+
+	EC2MetadataRequestTimeout = time.Duration(1 * time.Second)
```

Comment:
Legacy code. I can only guess. I think the reason would be to have all timeouts in one place so that we can easily change it if needed